core/src/dunit/scala/io/snappydata/cluster/SecureSplitClusterDUnitTest.scala

```diff
@@ -0,0 +1,325 @@
+  // create, drop, alter tables; select, insert, update, delete rows
+  //
+  // udf install jar
+
+  // create row and column table in embedded mode and select, insert, update, delete and
+  // drop from smart side and vice versa.
+  def testSQLOpsWithValidCredentials(): Unit = {
+    val props = new Properties()
+    props.setProperty(Attribute.USERNAME_ATTR, jdbcUser1)
+    props.setProperty(Attribute.PASSWORD_ATTR, jdbcUser1)
+    val conn = SplitClusterDUnitTest.getConnection(locatorClientPort, props)
+    val stmt = conn.createStatement()
+
+    // create row and column table in embedded mode
+    SplitClusterDUnitTest.createTableUsingJDBC(eColTab1, "column", conn, stmt, Map.empty, false)
+    SplitClusterDUnitTest.createTableUsingJDBC(eRowTab1, "row", conn, stmt, Map.empty, false)
+
+
+    val snc = testObject.getSnappyContextForConnector(locatorClientPort, props)
+
+    // insert
+    snc.sql(s"insert into $eColTab1 values (10000, 'ten thousand', 1000.23)")
+    snc.sql(s"insert into $eRowTab1 values (10000, 'ten thousand', 1000.23)")
+    // select
+    var count = snc.sql(s"select count(*) from $eColTab1").collect()(0).getLong(0)
+    assert(count == 1, s"expected 1 rows but found $count in $eColTab1")
+    count = snc.sql(s"select count(*) from $eRowTab1").collect()(0).getLong(0)
+    assert(count == 1, s"expected 1 rows but found $count in $eRowTab1")
+
+    // update
+    snc.sql(s"update $eColTab1 set col1 = 5000 where col1 = 10000")
+    snc.sql(s"update $eRowTab1 set col1 = 5000 where col1 = 10000")
+    var col1 = snc.sql(s"select * from $eColTab1").collect()(0).get(0)
+    assert(col1 == 5000, s"Update failed in $eColTab1, found value $col1")
+    col1 = snc.sql(s"select * from $eRowTab1").collect()(0).get(0)
+    assert(col1 == 5000, s"Update failed in $eRowTab1, found value $col1")
+
+    // delete
+    snc.sql(s"delete from $eColTab1 where col1 = 5000")
+    snc.sql(s"delete from $eRowTab1 where col1 = 5000")
+
+    var rows = snc.sql(s"select * from $eColTab1").collect().length
+    assert(rows == 0, s"expected 0 rows but found $rows in $eColTab1")
+    rows = snc.sql(s"select * from $eRowTab1").collect().length
+    assert(rows == 0, s"expected 0 rows but found $rows in $eRowTab1")
+
+    // insert from embedded side. Adds 1005 rows each
+    SplitClusterDUnitTest.populateTable(eColTab1, conn)
+    SplitClusterDUnitTest.populateTable(eRowTab1, conn)
+
+    rows = snc.sql(s"select * from $eColTab1").collect().length
+    assert(rows == 1005, s"expected 1005 rows but found $rows in $eColTab1")
+    rows = snc.sql(s"select * from $eRowTab1").collect().length
+    assert(rows == 1005, s"expected 1005 rows but found $rows in $eRowTab1")
+
+    // drop
+    snc.sql(s"drop table $eColTab1")
+    snc.sql(s"drop table $eRowTab1")
+    assert(!snc.sparkSession.catalog.tableExists(eColTab1), s"$eColTab1 not dropped")
+    assert(!snc.sparkSession.catalog.tableExists(eRowTab1), s"$eRowTab1 not dropped")
+
+    // vice versa
+  }
```

Comment:
Please add test case where user tries to update tables in hive metastore and it should fail.
Also test cases for alter table scenario.